gen/gen.go

```diff
@@ -51,7 +50,7 @@ type Header struct {
 
 // String generates the header for cases_test.go file.
 func (h Header) String() string {
-	return fmt.Sprintf("// Source: %s\n// Commit: %s", h.Origin, h.Commit)
+	return fmt.Sprintf("// AUTOGENERATED FILE: do not change manually (how to run generator: https://github.com/exercism/go#synchronizing-tests-and-instructions)\n// Source: %s\n// Commit: %s\n", h.Origin, h.Commit)
```

Comment:
Maybe the text can be made a bit smoother. Also the line length is rather long for the new comment. Here a potential alternative:
```
// This is an auto-generated file. Do not change it manually. Run the generator to update the file.
// See https://github.com/exercism/go#synchronizing-tests-and-instructions.
```